api/lib/domain/services/algorithm-methods/flash.js

```diff
@@ -162,6 +174,15 @@ function _filterAlreadyAnsweredCompetences({ answers, challenges, forcedCompeten
   );
 }
 
+function _getLastAnswersCompetenceIds(allAnswers, allChallenges, numberOfAnswers) {
+  const lastAnswers = allAnswers.slice(-numberOfAnswers);
+  const competenceIds = lastAnswers
+    .map((answer) => _findChallengeForAnswer(allChallenges, answer))
+    .map(({ competenceId }) => competenceId);
```

Comment:
Un peu dommage de parcourir 2 fois
```suggestion
    .map((answer) => _findChallengeForAnswer(allChallenges, answer)).competenceId;
```
ou 
```suggestion
    .map((answer) => {
    const { competenceId } = _findChallengeForAnswer(allChallenges, answer))
    return competenceId;
```